src/web/mjs/connectors/VizShonenJump.mjs

```diff
@@ -34,26 +36,87 @@ export default class VizShonenJump extends Connector {
+        });
+    }
+
+    async _getMangasAvailibleByChapters() {
         const request = new Request(new URL('/shonenjump', this.url), this.requestOptions);
         let data = await this.fetchDOM(request);
 
         if ( !data.innerText.includes('Latest free chapters') ) {
             alert('This website is geolocked. It can only be accessed from the USA.\nYou may use MANGA Plus instead.', this.label, 'info');
             return;
         }
-
-        data = [...data.querySelectorAll('div.o_sort_container div > a')];
+        
+        data = [...data.querySelectorAll('a.o_chapters-link')];
 
         return data.map(manga => {
             return {
-                id: manga.pathname,
-                title: manga.querySelector(':nth-child(2)').innerText.trim()
+                id: JSON.stringify({
+                    id: manga.pathname,
+                    title: manga.innerText.trim(),
+                    entry_type: this.EntryTypeEnum.chapter
+                }),
+                title: manga.innerText.trim(),
             };
         });
     }
 
-    async _getChapters(manga) {
+    async _getMangas() {
+        const mangaAvailibleByVolumes = await this._getMangasAvailibleByVolumes();
+        var mangaAvailibleByChapters = await this._getMangasAvailibleByChapters();
+
+        mangaAvailibleByVolumes.forEach(mangaAvailibleByVolume => {
+                const index = mangaAvailibleByChapters.findIndex(element => element.title === mangaAvailibleByVolume.title);
+                // Found it.
+                if (index != -1) {
+                    var complexManga = JSON.parse(mangaAvailibleByChapters[index].id);
+
+                    mangaAvailibleByChapters[index] = {
+                        id: JSON.stringify({
+                            id: complexManga.id,
+                            volume_id: mangaAvailibleByVolume.volume_id,
+                            title: mangaAvailibleByVolume.title,
+                            entry_type: this.EntryTypeEnum.volume | this.EntryTypeEnum.chapter
+                        }),
+                        title: mangaAvailibleByVolume.title,
+                    };
+                }
+                // Didn't find it, just add this manga straight into the list.
+                else {
+                    mangaAvailibleByChapters.push({
+                        id: JSON.stringify({
+                            volume_id: mangaAvailibleByVolume.volume_id,
+                            id: mangaAvailibleByVolume.volume_id,
+                            title: mangaAvailibleByVolume.title,
+                            entry_type: this.EntryTypeEnum.volume
+                        }),
+                        title: mangaAvailibleByVolume.title,
+                    });
+                }
+            })
+
+        return mangaAvailibleByChapters;
+    }
+
+    async _getMangaChapters(manga) {
+        if ((manga.entry_type & this.EntryTypeEnum.chapter) != this.EntryTypeEnum.chapter) {
```

Comment:
This is used as a Flag enum, so it's not one or the other, it could be both. A user might have manga chapters available via SJ, as well as volumes they've purchased from Viz. So a switch case doesn't work here, as if both flags were set, the enums value would be `3` and neither condition would be hit.

Thus we union on the values returned from both functions, but sometimes one or the other might be empty.

Reply:
yup this was just be being kinda dumb
`[1,3].includes(this.EntryTypeEnum.chapter)` 
is probably more readable, using bitwise is rather uncommon in js